### PR TITLE
Streamline dockerfiles

### PR DESCRIPTION
**Pull request by wpears**
Changes to both dockerfiles herein. Main changes:

* api container no longer builds node from source
* mongo container loads data on startup (though the context needed to be expanded to allow run the reloading script

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,20 +1,20 @@
...
 ADD . /usr/local/node/hmda-edit-check-api
 WORKDIR /usr/local/node/hmda-edit-check-api
 
 # Set the environment
-ENV NODE_VERSION=v0.12.5 NODE_ENV=production PORT=8000 NODE_TLS_REJECT_UNAUTHORIZED=0
+ENV NODE_ENV=production PORT=8000 NODE_TLS_REJECT_UNAUTHORIZED=0
 
-# Use a custom build script instead of messy chained together RUN
-# or multiple RUN statements that add bloat to the image
+#
```
Comment (hkeeler) [inline Dockerfile]: Looks like there's supposed to be a comment here?

File: mongo-docker/start.sh [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+# Start mongo
+/tmp/entrypoint.sh mongod&
```
Comment (hkeeler) [inline mongo-docker/start.sh]: I'm curious how Docker handles this, with the `mongod` processes it put into the background.
* Do `mongod` logs get written to stdout/stderr and get picked up as Docker logs as when running in the foreground?
* Does the container stop when the `mongod` process dies?

Do `mongod` logs get logged to the console appropriately?  I know there can be issues when you put a process into the background.

File: mongo-docker/start.sh [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+
+sleep 5
+
+# Load data
+node /usr/local/app/data/reload_mongo.js
```
Comment (hkeeler) [inline mongo-docker/start.sh]: So, will there be a time where the service is up, but not all data will be loaded yet?  Does it load fast enough that we won't care?

File: mongo-docker/Dockerfile [hunk 4]
```diff
@@ -1,18 +1,18 @@
...
-# https://github.com/docker-library/mongo/blob/master/2.6/Dockerfile
-FROM alpine:3.2
+FROM mhart/alpine-node:0.10
 
-# Use a custom build script instead of messy chained together RUN
-# or multiple RUN statements that add bloat to the image
-ADD run.sh /tmp/run.sh
+ENV NODE_ENV=production
+
+COPY mongo-docker /tmp
```
Comment (hkeeler) [inline mongo-docker/Dockerfile]: This `Dockerfile` is in the `mongo-docker` directory.  I believe all `COPY` commands are relative to the location of the `Dockerfile`.  Does this work?

File: mongo-docker/start.sh (hunk 2 above)
Comment (wpears) [inline mongo-docker/start.sh]: Nope, nope, and nope. This approach is prettttty hacky and in fact needs to be replaced (before a merge, actually).

File: mongo-docker/start.sh (hunk 3 above)
Comment (wpears) [inline mongo-docker/start.sh]: I think the answer to this is yes; typically loads in 5-10 seconds.. though imo this is still not ideal. Though I think a solution just occurred to me that will fix both this and the above

File: mongo-docker/Dockerfile (hunk 4 above)
Comment (wpears) [inline mongo-docker/Dockerfile]: I think COPY commands are relative to the build context (so whatever you pass to `docker build` while the location of the Dockerfile can be provided with the `-f` flag. The reallll question is will this play nicely with Jenkins :bomb:

**Comment by hkeeler** [comment]
Cool.  Yeah, this looks better.  Merging...

**Pull request merged**